premailer/tests/test_premailer.py

```diff
@@ -2984,3 +2984,57 @@ def test_pseudo_selectors_without_selector(self):
         p = Premailer(html, exclude_pseudoclasses=False, keep_style_tags=True)
         result_html = p.transform()
         compare_html(expect_html, result_html)
+
+    def test_preserve_handlebar_syntax_false(self):
+        """Demonstrate encoding of handlebar syntax with preservation disabled.
+
+        Original issue: https://github.com/peterbe/premailer/issues/248
+        """
+
+        html = """
+            <html>
+            <img src="{{ img_src }}">
+            <a href="{{ url }}"></a>
+            </html>
+        """
+
+        expect_html = """
+<html>
+    <head>
+    </head>
+    <body>
+    <img src="%7B%7B%20img_src%20%7D%7D">
+    <a href="%7B%7B%20url%20%7D%7D"></a>
+    </body>
+</html>
+        """
+        p = Premailer(html)
+        result_html = p.transform()
+        compare_html(expect_html, result_html)
+
+    def test_preserve_handlebar_syntax_true(self):
+        """Demonstrate encoding of handlebar syntax with preservation enabled.
+
+        Original issue: https://github.com/peterbe/premailer/issues/248
+        """
+
+        html = """
+            <html>
+            <img src="{{ img_src }}">
+            <a href="{{ url }}"></a>
```

Comment:
This second example doesn't exercise the importance of using `urllib.parse.unquote`. Perhaps...
```suggestion
            <a href="{{ url('some thing') }}"></a>
```

Note, I know next to nothing about Handlebar.

Reply:
Including more complex content is a great idea. I discovered a problem when attempting to include `"` in the handlebars. My solution was to replace `"` within the handlebars with `%22` before being encoded so that when they were decoded they would come back out as normal `"`. Changes here: https://github.com/peterbe/premailer/pull/252/commits/03c0db4dc242a4b66f49facf55dfbb3a3a5bc575

Reply:
I got the test data from https://docs.djangoproject.com/en/3.1/ref/templates/language/#string-literals-and-automatic-escaping because it seemed to be the most complex example of handlebar syntax that I could find.